R/redist.R

```diff
@@ -628,37 +629,676 @@ redist.combine.anneal <- function(file_name){
+                algout <- list()
+
+            }else{
+                cds <- preprocout$data$initcds
+                ## Initialize algout object (for use in ecutsAppend)
+                algout <- list()
+                ## Temperature Adjacency Matrix
+                if(adjswaps){
+                  tempadjMat <- tempadj
+                }
+            }
+
+        }
+
+        #######################
+        ## Run the algorithm ##
+        #######################
+        for(j in 1:length(nsimsAdj)){
+
+            cat("Swap ", j, "\n", append = TRUE)
+            ## Run algorithm
+            temp <- swMH(aList = preprocout$data$adjlist,
+                         cdvec = cds,
+                         cdorigvec = preprocout$data$initcds,
+                         popvec = preprocout$data$popvec,
+                         grouppopvec = preprocout$data$grouppopvec,
+                         nsims = nsimsAdj[j],
+                         eprob = eprob,
+                         pct_dist_parity = preprocout$params$pctdistparity,
+                         beta_sequence = preprocout$params$betaseq,
+                         beta_weights = preprocout$params$betaweights,
+                         ssdmat = preprocout$data$ssdmat,
+                         lambda = lambda,
+                         beta_population = betapop,
+                         beta_compact = betacompact,
+                         beta_segregation = betaseg,
+                         beta_similar = betasimilar,
+                         anneal_beta_population = preprocout$params$temperbetapop,
+                         anneal_beta_compact = preprocout$params$temperbetacompact,
+                         anneal_beta_segregation = preprocout$params$temperbetaseg,
+                         anneal_beta_similar = preprocout$params$temperbetasimilar,
+                         adjswap = preprocout$params$adjswaps)
+
+            ## Combine data
+            algout <- ecutsAppend(algout,temp)
+
+            ## Get temperature
+            beta <- temp$beta_sequence[nsimsAdj[j]]
+
+            ## Get likelihood
+            if(constraint == "compact"){
+                like <- exp(temp$constraint_compact[nsimsAdj[j]])
+            }
+            else if(constraint == "population"){
+                like <- exp(temp$constraint_pop[nsimsAdj[j]])
+            }
+            else if(constraint == "segregation"){
+                like <- exp(temp$constraint_segregation[nsimsAdj[j]])
+            }
+            else{
+                like <- exp(temp$constraint_similar[nsimsAdj[j]])
+            }
+            cat("Likelihood is", like, "\n", append = TRUE)
+            ## Use MPI to exchange swap information
+            ##
+            ## Tag guide: 1 -> likelihood sent
+            ##            2 -> temperature sent
+            ##            3 -> acceptance probability sent
+            ##
+
+            if(adjswaps){
+
+                cat("Start adjswaps\n", append = TRUE)

```

Comment:
I used these comments to debug the MPI function, which I've found helpful since the MPI code can be a bit brittle. I'd recommend keeping them in for your own sake, but maybe with a `debug = TRUE/FALSE` argument.